### PR TITLE
Fix race condition when receiving http response on Native

### PR DESCRIPTION
**Pull request by marclefrancois**
## Description
Make sure we use a frozen publisher to return the response and that it is frozen before the call is started

## Motivation and Context
The BehaviourSubject we are using to return the response was frozen only after starting the request. This created a possible race condition where the response was assigned to the value of the publisher from another thread while the publisher is being frozen.

This caused various  crashes in the Kotlin/Native freezing process since it is not thread safe

## How Has This Been Tested?
Tested in one of our application that was able to reproduce the problem by spamming the same http request that was hitting the local cache

## Types of changes
- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: http/build.gradle.kts [hunk 1]
```diff
@@ -40,7 +40,7 @@ kotlin {
         val commonMain by getting {
             dependencies {
                 implementation("com.mirego.trikot:trikotFoundation:${project.extra["trikot_foundation_version"]}")
-                implementation("com.mirego.trikot:streams:${project.extra["trikot_foundation_version"]}")
+                implementation("com.mirego.trikot:streams:${project.extra["trikot_streams_version"]}")
```
Comment (marcantoinefortier) [inline http/build.gradle.kts:43]: 👀

**Review by marcantoinefortier — APPROVED** [review]
🎉

**Review by guillaumeaudet — APPROVED** [review]
💯

**Review by martingagnon — COMMENTED** [review]
Late 👍

**Pull request merged**